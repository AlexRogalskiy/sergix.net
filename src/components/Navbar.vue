<template lang="pug">
nav.navbar
  g-link(to='/')
    img(src='@/assets/image/logo/sergix-logo.svg'
      alt='Sergix').nav_logo
  .nav_link_list
    g-link.nav_link(to='/about') ABOUT
    g-link.nav_link(to='/design') DESIGN
    g-link.nav_link(to='/projects') PROJECTS
    g-link.nav_link(to='/mode') MODE
    a.nav_link(href='https://github.com/sergix') GITHUB
  button(
    type='button'
    aria-label='Menu'
    aria-controls='navigation'
    v-bind:class='{ "is-active": menuButtonActive }'
    v-on:click='menuButtonActive = !menuButtonActive').menu-icon.hamburger.hamburger--minus
    span.hamburger-box
      span.hamburger-inner
  NavMenu(:active="menuButtonActive")
</template>

<script>
import NavMenu from '@/components/NavMenu'

export default {
  name: 'Navbar',
  components: {
    NavMenu,
  },
  data: () => ({
    menuButtonActive: false,
  }),
}
</script>

<style lang="scss" scoped>
.navbar {
  align-items: center;
  background-color: $light-color;
  display: flex;
  justify-content: space-between;
  padding: 2vh;
  z-index: 1;

  .nav_logo {
    float: left;
    width: 10em;
  }

  .nav_link_list {
    float: right;
  }

  .nav_link {
    color: $dark-color;
    font-size: 1rem;
    margin: 2rem;
    text-decoration: none;
    padding: 0.5rem;
  }

  .menu-icon {
    margin-right: -1rem;
    z-index: 1000;
  }
}

@media screen and (orientation: portrait) {
  .nav_link_list {
    display: none;
  }

  .menu-icon {
    display: initial;
  }
}

@media screen and (orientation: landscape) and (min-width: $responsive-sm) {
  .nav_link_list {
    display: initial;
  }

  .menu-icon {
    display: none;
  }
}

@media screen and (orientation: landscape) and (max-width: $responsive-sm) {
  .nav_link_list {
    display: none;
  }

  .menu-icon {
    display: initial;
  }
}

@media screen and (max-width: $responsive-sm) {
  .navbar {
    margin: 0;
    padding: 2vh 5vh 2vh 5vh;
  }
}
</style>
