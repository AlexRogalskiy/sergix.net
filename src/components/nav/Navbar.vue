<template>
  <nav class="flex flex-row flex-shrink-0 h-24 items-center">
    <MobileMenu :active="menuButtonActive" @close="closeMenu" />
    <g-link class="logo" v-show="!menuButtonActive" to="/">
      <img class="w-48 ml-2 pb-2" src="@/assets/image/logo/sergix.svg" />
    </g-link>
    <ul
      class="hidden md:flex flex-row mt-1 flex-1 items-center ml-8"
      v-show="!isIndexPage"
    >
      <li>
        <g-link to="/websites">WEBSITES</g-link>
      </li>
      <li>
        <g-link to="/designs">DESIGNS</g-link>
      </li>
      <li>
        <g-link to="/blog">BLOG</g-link>
      </li>
    </ul>
    <button
      class="btn btn-gradient-outline ml-auto mr-6 pt-button-fix md:hidden"
      :class="{ hidden: isIndexPage }"
      @click="toggleMenu"
    >
      {{ menuButtonActive ? 'CLOSE MENU' : 'MENU' }}
    </button>
  </nav>
</template>

<script>
import MobileMenu from '@/components/nav/MobileMenu'

export default {
  name: 'Navbar',
  components: {
    MobileMenu,
  },
  data() {
    return {
      menuButtonActive: false,
    }
  },
  computed: {
    isIndexPage() {
      return this.$route.path === '/'
    },
  },
  methods: {
    toggleMenu() {
      this.menuButtonActive = !this.menuButtonActive
      this.toggleBodyOverflow()
    },
    toggleBodyOverflow() {
      // no ssr
      if (process.isClient) {
        window.scrollTo(0, 0)
        const el = document.body
        const className = 'overflow-hidden'
        this.menuButtonActive
          ? el.classList.add(className)
          : el.classList.remove(className)
      }
    },
    closeMenu() {
      this.menuButtonActive = false
      this.toggleBodyOverflow()
    },
  },
}
</script>

<style lang="sass" scoped>
.logo
  transition: display 0.5s ease-in
  margin-top: .85rem

ul > li
  @apply px-12

  & > a
    &::after
      content: ''
      height: 2px
      transition: width 0.2s
      transition-delay: 0.1s
      @apply w-0
      @apply mx-auto
      @apply block
      @apply bg-primary

    &.active--exact::after
      @apply w-full

    &:hover::after
      @apply w-1/2
</style>
