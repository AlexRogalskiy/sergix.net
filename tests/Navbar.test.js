import { mount, RouterLinkStub } from '@vue/test-utils'
import Navbar from '@/components/Navbar'

describe('component/Navbar', () => {
  let wrapper = mount(Navbar, {
    stubs: {
      'g-link': RouterLinkStub,
    },
  })

  it('renders correctly', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })
})
