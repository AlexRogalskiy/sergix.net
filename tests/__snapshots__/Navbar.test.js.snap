// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component/Navbar renders correctly 1`] = `
<nav class="navbar"><a><img src="@/assets/image/logo/sergix-box-logo.svg" alt="Sergix" class="nav_logo"></a>
  <li class="nav_link_list"><a class="nav_link">ABOUT</a><a class="nav_link">DESIGN</a><a class="nav_link">PROJECTS</a><a class="nav_link">MODE</a><a href="https://github.com/sergix" class="nav_link">GITHUB</a></li>
</nav>
`;
